kaldb/src/test/java/com/slack/kaldb/metadata/zookeeper/CachedMetadataStoreImplTest.java

```diff
@@ -100,6 +109,67 @@ public void tearDown() throws IOException {
     return cachedMetadataStore;
   }
 
+  @Test
+  public void raceCondition() throws Exception {
```

Comment:
Please use a more descriptive name for this test like testRaceConditionDuringClose.

Reply:
Ah, I overlooked going back and fixing that. It's now `raceConditionOnMetadataStoreCloseListeners`